pyogrio/tests/test_geopandas_io.py

```diff
@@ -139,6 +143,49 @@ def test_read_datetime(test_fgdb_vsi):
         assert df.SURVEY_DAT.dtype.name == "datetime64[ns]"
 
 
+def test_read_datetime_tz(test_datetime_tz, tmp_path):
+    df = read_dataframe(test_datetime_tz)
+    raw_expected = ["2020-01-01T09:00:00.123-05:00", "2020-01-01T10:00:00-05:00"]
+
+    if Version(pd.__version__) >= Version("2.0.0"):
+        expected = pd.to_datetime(raw_expected, format="ISO8601").as_unit("ms")
+    else:
+        expected = pd.to_datetime(raw_expected)
+    expected = pd.Series(expected, name="col")
+
+    assert_series_equal(df.col, expected)
+    # test write and read round trips
+    # TODO gpkg doesn't work here, at least for my local gdal, writes NaT
+    fpath = tmp_path / "test.geojson"
+    write_dataframe(df, fpath)
+    df_read = read_dataframe(fpath)
+    assert_series_equal(df_read.col, expected)
+
+
+def test_write_datetime_mixed_offset(tmp_path):
+    # Summer Time (GMT+11), standard time (GMT+10)
+    dates = ["2023-01-01 11:00:01.111", "2023-06-01 10:00:01.111"]
+    tz = pytz.timezone("Australia/Sydney")
+    ser_naive = pd.Series(pd.to_datetime(dates), name="dates")
+    ser_localised = ser_naive.dt.tz_localize(tz)
+    df = gp.GeoDataFrame(
+        {"dates": ser_localised, "geometry": [Point(1, 1), Point(1, 1)]}
+    )
+    fpath = tmp_path / "test.geojson"
+    write_dataframe(df, fpath)
+    df_no_tz = read_dataframe(
+        fpath, datetime_as_string=False
+    )  # TODO this shouldn't be called datetime as string in the pandas layer,
+    #     should it even be accessible?
```

Comment:
Unless we have some users asking for it, I wouldn't expose it in read_dataframe (so only in raw.read), just to keep it simpler in read_dataframe